test/core/ticktree/TickTreeUpdateOffer.t.sol

```diff
@@ -0,0 +1,347 @@
+        run_update_offer_scenario(
+          UpdateOfferScenario({
+            tickScenario: tickScenario,
+            newTick: tickScenario.tick,
+            offerTickListSize: tickListScenario[0],
+            offerPos: tickListScenario[1]
+          }),
+          false
+        );
+        if (tickScenario.hasHigherTick) {
+          run_update_offer_scenario(
+            UpdateOfferScenario({
+              tickScenario: tickScenario,
+              newTick: tickScenario.higherTick,
+              offerTickListSize: tickListScenario[0],
+              offerPos: tickListScenario[1]
+            }),
+            false
+          );
+        }
+        if (tickScenario.hasLowerTick) {
+          run_update_offer_scenario(
+            UpdateOfferScenario({
+              tickScenario: tickScenario,
+              newTick: tickScenario.lowerTick,
+              offerTickListSize: tickListScenario[0],
+              offerPos: tickListScenario[1]
+            }),
+            false
+          );
+        }
+      }
+    }
+    vm.resumeGasMetering();
+  }
+
+  // function test_single_update_offer_scenario() public {
+  //   run_update_offer_scenario(
+  //     UpdateOfferScenario({
+  //       tickScenario: TickScenario({
+  //         tick: 0,
+  //         hasHigherTick: true,
+  //         higherTick: 524287,
+  //         higherTickListSize: 1,
+  //         hasLowerTick: true,
+  //         lowerTick: -16384,
+  //         lowerTickListSize: 0
+  //       }),
+  //       offerTickListSize: 1,
+  //       offerPos: 0,
+  //       newTick: -16384
+  //     }),
+  //     true
+  //   );
+  // }
+
+  function run_update_offer_scenario(UpdateOfferScenario memory scenario, bool printToConsole) internal {
+    // NB: Enabling all console.log statements will trigger an out-of-memory error when running through all test scenarios.
+    // `printToConsole` is used to enable logging for specific scenarios.
+
+    if (printToConsole) {
+      console.log("update offer scenario");
+      console.log("  oldTick: %s", toString(Tick.wrap(scenario.tickScenario.tick)));
+      console.log("  newTick: %s", toString(Tick.wrap(scenario.newTick)));
+      console.log("  offerTickListSize: %s", scenario.offerTickListSize);
+      console.log("  offerPos: %s", scenario.offerPos);
+      if (scenario.tickScenario.hasHigherTick) {
+        Tick higherTick = Tick.wrap(scenario.tickScenario.higherTick);
+        console.log("  higherTick: %s", toString(higherTick));
+        console.log("  higherTickListSize: %s", vm.toString(scenario.tickScenario.higherTickListSize));
+      }
+      if (scenario.tickScenario.hasLowerTick) {
+        console.log("  lowerTick: %s", toString(Tick.wrap(scenario.tickScenario.lowerTick)));
+        console.log("  lowerTickListSize: %s", vm.toString(scenario.tickScenario.lowerTickListSize));
+      }
+    }
+
+    // 1. Capture VM state before scenario so we can restore it after
+    uint vmSnapshotId = vm.snapshot();
```

Comment:
This is unreliable, right? Alternatives:
- Actually create a ton of tests (also alleviates the memory leak problems in loops)
- Still loop, but create a new Mangrove every time, so no need to reset a state

Reply:
I think it is only unreliable when tests run out of gas/memory. When the tests work, they do so reliably.

I think we should keep it as-is for now and move on. If we need to update/change the tick tree tests at some point, we can look into improving this. Ok?